src/ipfs.ts

```diff
@@ -0,0 +1,53 @@
+import { Readable } from 'stream';
+
+import { CID } from 'ipfs-http-client';
+
+import { toString } from './merkle.js';
+import { shared as Shared } from './shared.js';
+import { MerkleTree } from './types.js';
+
+// TODO: local IPFS node required for the development and tests
+export async function uploadTree(t: MerkleTree): Promise<CID> {
+    const data = toString(t);
+
+    return Shared.IPFS.pinFileToIPFS(Readable.from(data), {
+        pinataMetadata: {
```

Comment:
should we store some oracle info or not really?

Reply:
in case some oracle suddenly built another tree and pinned it to the ipfs, when do we know this? will it be just an unused file here?

Reply:
Correct. His report won't reach consensus. Pinning service and local artifacts may be inspected to find an issue and unpin the file.